packages/auth/src/core/strategies/email_and_password.ts

```diff
@@ -319,7 +327,11 @@ export async function createUserWithEmailAndPassword(
     clientType: RecaptchaClientType.WEB
   };
   let signUpResponse: Promise<IdTokenResponse>;
-  if (authInternal._getRecaptchaConfig()?.emailPasswordEnabled) {
+  if (
+    authInternal
+      ._getRecaptchaConfig()
+      ?.isProviderEnabled(RecaptchaProvider.EMAIL_PASSWORD_PROVIDER)
+  ) {
     const requestWithRecaptcha = await injectRecaptchaFields(
```

Comment:
FYI There is a somewhat related PR in flight - https://github.com/firebase/firebase-js-sdk/pull/7666

Reply:
ACK. Yes, I'm aware of that PR and I'm happy to resolve the conflicts here if needed.

Reply:
@NhienLam https://github.com/firebase/firebase-js-sdk/pull/7666 is now merged. Please resolve the conflicts when you get a chance. Thanks!